metaflow/plugins/datatools/s3/s3.py

```diff
@@ -819,6 +819,75 @@ def _head():
 
         return list(starmap(S3Object, _head()))
 
+    def delete(self, key):
+        """
+        Delete a single object in S3.
+        Parameters
+        ----------
+        key : str
+            Object to delete. It can be an S3 url or a path suffix.
+        """
+
+        url = self._url(key)
+        src = urlparse(url)
+
+        def _delete(s3, tmp):
+            if not src.path.lstrip("/"):
+                raise TypeError("'None' type object cannot be deleted.")
+            else:
+                return s3.delete_object(Bucket=src.netloc, Key=src.path.lstrip("/"))
+
+        self._one_boto_op(_delete, url, create_tmp_file=False)
+
+    def delete_many(self, keys):
+        """
+        Delete multiple objects from s3
+        Parameters
+        ----------
+        keys: List[str]
+            List of keys to delete.
+        """
+
+        urls = [self._url(key) for key in keys]
+        src = [urlparse(u) for u in urls]
+
+        Bucket_names = set([s.netloc for s in src])
+        if len(Bucket_names) > 1:
+            raise Exception("Cannot delete from multiple buckets at once")
```

Comment:
We may want to support this to be consistent with the rest. In that case, maybe we can do this operation over buckets using s3op?